pkg/discovery/discover_test.go

```diff
@@ -0,0 +1,548 @@
+			PollInterval: 1 * time.Second,
+		},
+	}
+
+	mc := &mockClient{}
+	mc.reqs = make(map[string]*api.Response)
+	for k, v := range mapOfResponses {
+		mc.reqs[k] = v
+	}
+
+	apClient := anypoint.NewClient(ac.MulesoftConfig, anypoint.SetClient(mc))
+	agent := New(ac, apClient)
+
+	return agent, mc
+}
+
+func TestGetServiceDetailFailsWhenGetPoliciesFails(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:          456,
+		EndpointURI: "google.com",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestGetServiceDetailFailsWhenGetExchangeAssetFails(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestGetServiceDetailWhenExchangeAssetSpecFileIsEmpty(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"Files\": [{}]}"),
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	sd, _ := agent.getServiceDetail(asset, a)
+	assert.Nil(t, sd)
+}
+
+func TestGetServiceDetailFailsWhenGetSpecFromExchangeFileFails(t *testing.T) {
+	agent, mc := buildAgentWithCustomMockAnypointClient()
+
+	mc.reqs["/apimanager/api/v1/organizations/333/environments/111/apis/456/policies"] = &api.Response{
+		Code: 200,
+		Body: []byte(`[{}]`),
+	}
+
+	mc.reqs["/exchange/api/v2/assets/123/456/89"] = &api.Response{
+		Code: 200,
+		Body: []byte("{\"Files\": [{\"Classifier\":\"oas\",\"Generated\":false, \"ExternalLink\": \"https://localhost/swagger.json\"}]}"),
+	}
+
+	mc.reqs["https://localhost/swagger.json"] = &api.Response{
+		Code: 400,
+		Body: nil,
+	}
+
+	asset := &anypoint.Asset{}
+
+	a := &anypoint.API{
+		ID:           456,
+		EndpointURI:  "google.com",
+		GroupID:      "123",
+		AssetID:      "456",
+		AssetVersion: "89",
+	}
+
+	_, err := agent.getServiceDetail(asset, a)
+	assert.NotNil(t, err)
+}
+
+func TestGetServiceDetailFailsWhenGetExchangeAssetIconFails(t *testing.T) {
```

Comment:
If an error is thrown in one of the earlier requests will `getServiceDetail` return that error?